src/components/MarketMySharesForm/index.js

```diff
@@ -87,7 +82,10 @@ class MarketMySharesForm extends Component {
   @autobind
   handleSellShare(shareId, shareAmount) {
     const shareIndex = this.props.marketShares.map(share => share.id).indexOf(shareId)
-    return this.props.sellShares(this.props.market, shareIndex, shareAmount).then(() => this.props.reset())
+    const shareBalance = new Decimal(this.props.marketShares[shareIndex].balance).div(1e18).toDP(2, 1)
+    const selectedSellAmount = new Decimal(shareAmount)
```

Comment:
This in Wei now, yes? ^^

Reply:
@andre-meyer share.balance is in Wei, shareAmount is in ether

Reply:
But then it does `shareBalance (in wei) - shareAmountToSell (in ether)`, right? I think I'm confused

Reply:
@andre-meyer `const shareBalance = new Decimal(this.props.marketShares[shareIndex].balance).div(1e18)`
It is divided before subtraction